share/smack/top.py

```diff
@@ -296,6 +297,12 @@ def smack_headers():
 def smack_lib():
   return os.path.join(smack_root(), 'share', 'smack', 'lib')
 
+def objc_clang_compile_command(args, lib = False):
+  cmd = default_clang_compile_command(args,lib)
+  objc_flags = try_command(['gnustep-config', '--objc-flags'])
```

Comment:
Is this part specific to Linux? it would be good to have something that works on OS X too, especially given that a Objective C is an Apple-centric technology.

Reply:
Yes, this part is specific to Linux. For macOS support, a different addition will be necessary. I agree that it will be useful, though, it's just not part of this PR

Reply:
I kind of understand why we need gnustep-config - to be able to determine which flags are needed to compile Objective C programs in Linux.
How different are Linux flags from OS X flags?

Reply:
From what I can tell, for this simple case, the default command that works for C will work, but if we are linking multiple files "-framework Foundation" may be required. Let me emphasize that I have NOT tested this.

Reply:
Check this out: https://stackoverflow.com/questions/110362/how-can-i-find-the-current-os-in-python
If you invoke `platform.system` in python, it is easy to find out if you are running in Linux or OS X.
So maybe that would be a relatively easy way to make this code cross-platform enabled. Thoughts?

Reply:
I've added this below... left the OS X section blank for now.